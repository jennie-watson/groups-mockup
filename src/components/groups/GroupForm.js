import React from "react";
import PropTypes from "prop-types";
import TextInput from "../common/TextInput";
import SelectInput from "../common/SelectInput";

const GroupForm = ({
  group,
  moderators,
  onSave,
  onChange,
  saving = false,
  errors = {}
}) => {
  return (
    <form onSubmit={onSave}>
      <h3 className="d-flex justify-content-center">
        {group.id ? "Edit" : "Create"} Group
      </h3>
      {errors.onSave && (
        <div className="alert alert-danger" role="alert">
          {errors.onSave}
        </div>
      )}
      <TextInput
        name="name"
        label="name"
        placeholder="Group Name"
        value={group.name}
        onChange={onChange}
        error={errors.name}
      />

      <textarea
        className="form-control"
        rows="3"
        name="description"
        label="description"
        placeholder="Group Description"
        value={group.description}
        onChange={onChange}
        error={errors.description}
      ></textarea>

      <br />
      <SelectInput
        className="moderator"
        name="moderatorUserId"
        value={group.moderatorUserId || ""}
        defaultOption="Select Moderator"
        options={moderators.map(moderator => ({
          value: moderator.id,
          text: moderator.name
        }))}
        onChange={onChange}
        error={errors.moderator}
      />

      {/* Radio buttons for Create Group Form */}
      <div className="form-check">
        <h5>Who can join your group?</h5>
        <input
          className="form-check-input"
          type="radio"
          name="neighbourhoodPrivacy"
          id="neighbourhoodPrivacy"
          value={group.neighbourhoodPrivacy}
          onChange={onChange}
          checked
        />
        <label className="form-check-label" htmlFor="neighbourhoodPrivacy">
          Anyone in my Neighbourhood
        </label>
      </div>

      <div className="form-check">
        <input
          className="form-check-input"
          type="radio"
          name="regionPrivacy"
          id="regionPrivacy"
          value={group.regionPrivacy}
          onChange={onChange}
        />
        <label className="form-check-label" htmlFor="regionPrivacy">
          Anyone in my Region
        </label>
      </div>

      <div className="form-check">
        <input
          className="form-check-input"
          type="radio"
          name="nzPrivacy"
          id="nzPrivacy"
          value={group.nzPrivacy}
          onChange={onChange}
        />
        <label className="form-check-label" htmlFor="nzPrivacy">
          Anyone in New Zealand
        </label>
      </div>

      <div className="form-check">
        <h5>Who can see your group&apos;s posts?</h5>
        <input
          className="form-check-input"
          type="radio"
          name="publicPost"
          id="publicPost"
          value={group.publicPost}
          onChange={onChange}
          checked
        />
        <label className="form-check-label" htmlFor="publicPost">
          Public
        </label>
        <p className="d-flex mb-2" style={{ fontSize: "small", opacity: 0.8 }}>
          Posts can be seen by the public
        </p>
      </div>

      <div className="form-check">
        <input
          className="form-check-input"
          type="radio"
          name="privatePost"
          id="privatePost"
          value={group.privatePost}
          onChange={onChange}
        />
        <label className="form-check-label" htmlFor="privatePost">
          Members only
        </label>
        <p className="d-flex mb-4" style={{ fontSize: "small", opacity: 0.8 }}>
          Only people who are members of the group can see the posts
        </p>
      </div>

      <button
        type="submit"
        disabled={saving}
        className="btn btn-primary rounded-lg d-flex mx-auto justify-content-center align-items-center w-50"
      >
        {saving ? "Creating..." : "Create a group"}
      </button>
    </form>
  );
};

GroupForm.propTypes = {
  moderators: PropTypes.array.isRequired,
  group: PropTypes.object.isRequired,
  errors: PropTypes.object,
  onSave: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  saving: PropTypes.bool
};

export default GroupForm;
