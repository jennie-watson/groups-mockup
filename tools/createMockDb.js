/* eslint-disable no-console */
const fs = require("fs");
const path = require("path");
const mockData = require("./mockData");

const { groups, moderators } = mockData;
const data = JSON.stringify({ groups, moderators });
const filepath = path.join(__dirname, "db.json");

fs.writeFile(filepath, data, function(err) {
  err ? console.log(err) : console.log("Mock DB created.");
});
