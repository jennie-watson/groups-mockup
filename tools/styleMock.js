// Mocks CSS imports for Jest. As suggested by https://jest.io/docs/en/webpack
module.exports = {};
