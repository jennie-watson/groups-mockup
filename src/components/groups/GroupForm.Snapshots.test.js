import React from "react";
import GroupForm from "./GroupForm";
import renderer from "react-test-renderer";
import { groups, moderators } from "../../../tools/mockData";

it("sets submit button label 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <GroupForm
      group={groups[0]}
      moderators={moderators}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving
    />
  );

  expect(tree).toMatchSnapshot();
});
it("sets submit button label 'Save' when saving is false", () => {
  const tree = renderer.create(
    <GroupForm
      group={groups[0]}
      moderators={moderators}
      onSave={jest.fn()}
      onChange={jest.fn()}
      saving={false}
    />
  );

  expect(tree).toMatchSnapshot();
});
