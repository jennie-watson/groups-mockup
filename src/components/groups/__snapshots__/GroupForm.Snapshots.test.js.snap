// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets submit button label 'Save' when saving is false 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Group
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="name"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="moderatorUserId"
    >
      Moderator
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="moderatorUserId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Moderator
        </option>
        <option
          value={1}
        >
          Cory House
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="description"
    >
      Description
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="description"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    type="submit"
  >
    Save
  </button>
</form>
`;

exports[`sets submit button label 'Saving...' when saving is true 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <h2>
    Edit
     Group
  </h2>
  <div
    className="form-group"
  >
    <label
      htmlFor="name"
    >
      Name
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="name"
        onChange={[MockFunction]}
        type="text"
        value="Securing React Apps with Auth0"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="moderatorUserId"
    >
      Moderator
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="moderatorUserId"
        onChange={[MockFunction]}
        value={1}
      >
        <option
          value=""
        >
          Select Moderator
        </option>
        <option
          value={1}
        >
          Cory House
        </option>
        <option
          value={2}
        >
          Scott Allen
        </option>
        <option
          value={3}
        >
          Dan Wahlin
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="description"
    >
      Description
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="description"
        onChange={[MockFunction]}
        type="text"
        value="JavaScript"
      />
    </div>
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Saving...
  </button>
</form>
`;
